### DISCUSSION TITLE
How to specify Token URL in swagger ui if you use multi OAuth2PasswordBearer

### DISCUSSION THREAD

**Discussion by yinzixie** [fastapi/fastapi#6478] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [ ] I commit to help with one of those options 👆

### Example Code

```python
admin_oauth2_schema = OAuth2PasswordBearer(
    tokenUrl=path.join(settings.ROOT_PATH + "/auth/admin-token")
)
candidate_oauth2_schema = OAuth2PasswordBearer(
    tokenUrl=path.join(settings.ROOT_PATH, "/auth/candidate-token")
)
```

### Description

In my app I have two token urls for different user. So how can I specify one of them in swagger ui? Currently, swagger ui will use one of the token url based on the last oauth Depends you used in your code. 

For example in my last router file I use depends `candidate_oauth2_schema` than the token url in swagger ui will become` /auth/candidate-token` but if I add another API below which uses depends `admin_oauth2_schema ` then the token url will change to "/auth/admin-token".

### Operating System

Windows

### FastAPI Version

0.70.1

### Python Version

3.9.0

**Comment by mh739025250** [comment]
I face the same problem. Does someone know how to work around it?

**Comment by rezmotaqi** [comment]
I'm having the same challenge. I have different authentication end points for normal user and admin user and I want to use 2 OAuth2PasswordBearer schemas which i have defined in separate schema modules, but only one of the (the one for admin) appears on Open api. Is there a solution here ? I am wondering maybe instead of having separate auth end points, i should user scopes for figuring out what kind of user is trying to login in order to do the appropriate lookup in database.

**Comment by dartt0n** [comment, marked as answer]
I have found the solution. You can specify different `scheme_name` for your `OAuth2PasswordBearers`.

Example:
```python
admin_oauth2_schema = OAuth2PasswordBearer(
    tokenUrl="/api/auth/admin-token",
    scheme_name="admin_oauth2_schema"
)
candidate_oauth2_schema = OAuth2PasswordBearer(
    tokenUrl="/api/auth/candidate-token",
    scheme_name="candidate_oauth2_schema"
)
```

This way, [Swagger UI](https://github.com/swagger-api/swagger-ui) will be able to different these two `OAuth2PasswordBearers` and will provide you useful UI to handle multiple oauth2 schemes.
![image](https://user-images.githubusercontent.com/62600293/159030249-92aecef5-271c-46bc-ad5d-7cf4116b4a00.png)
![image](https://user-images.githubusercontent.com/62600293/159030293-8db194d4-65f9-4c4d-8ec2-a21882a3d37e.png)

**Comment by yinzixie** [comment]
Thanks @Dartt0n it works.

**Comment by jiaulislam** [comment]
> I have found the solution. You can specify different `scheme_name` for your `OAuth2PasswordBearers`.
> 
> Example:
> 
> ```python
> admin_oauth2_schema = OAuth2PasswordBearer(
>     tokenUrl="/api/auth/admin-token",
>     scheme_name="admin_oauth2_schema"
> )
> candidate_oauth2_schema = OAuth2PasswordBearer(
>     tokenUrl="/api/auth/candidate-token",
>     scheme_name="candidate_oauth2_schema"
> )
> ```
> 
> This way, [Swagger UI](https://github.com/swagger-api/swagger-ui) will be able to different these two `OAuth2PasswordBearers` and will provide you useful UI to handle multiple oauth2 schemes. ![image](https://user-images.githubusercontent.com/62600293/159030249-92aecef5-271c-46bc-ad5d-7cf4116b4a00.png) ![image](https://user-images.githubusercontent.com/62600293/159030293-8db194d4-65f9-4c4d-8ec2-a21882a3d37e.png)

Thanks 👍🏻 Was scratching my head for hours lol.

**Comment by nikhil25803** [comment]
Hey @Dartt0n, does this still works? I tried, but it is not working ☹️
![image](https://github.com/tiangolo/fastapi/assets/93156825/d882ab32-f52c-43f6-aa83-3e05877b0994)

Only able to access one form in the UI
![image](https://github.com/tiangolo/fastapi/assets/93156825/1f527538-4e23-4682-aeee-5485dfd205ef)

**Reply by anhdt-work** [reply]
Me too T_T

**Reply by anhdt-work** [reply]
@tiangolo Can you check this? Thanks!

**Comment by Misnad** [comment]
Same issue here. Tried `scheme_name` and no luck.

**Comment by Misnad** [comment]
Turns out, I could resolve the issue by adding a route explicitly using the Admin model and get_current_admin dependency. This ensures that Swagger UI recognizes the security scheme and displays the “Authorize” button appropriately.


``` python
@router.get("/me")
async def read_users_me(current_user: Annotated[Bus, Depends(get_current_active_user)]):
    return current_user


@router.get("/boss")
async def read_admin_me(current_user: Annotated[Admin, Depends(get_current_acitve_admin)]):
    return current_user
```

**Reply by NarmadaM21** [reply]
is it working?....are you able to access two forms?

**Comment by AleksDominik** [comment]
The solution describe above still works. You need to inject the Annotated deps in one url so the swagger system notice the double OAuth2PasswordBearer. I made the same mistake because I start to implement the solution without having any url with with deps. 
Hope it could helps
